### PR TITLE
make luaPackages.sqlite work out of the box

### PR DESCRIPTION
**Pull request by teto**
## Description of changes

the tests worked because we exported LIBSQLITE. the user had to either set LIBSQLITE himself or set vim.g.sqlite_clib_path. This change provides a default if `vim.g.sqlite_clib_path` is unset such that it works out of the box. 

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by teto** [comment]
before this https://github.com/danielfalk/smart-open.nvim/ would complain about not being able to load sqlite even though I had installed the sqlite lua plugin via nix. After patching https://github.com/kkharji/sqlite.lua/blob/master/lua/sqlite/defs.lua, it works

**Review by GaetanLepage — APPROVED** [review]
LGTM

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 315732` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>lua54Packages.sqlite</li>
  </ul>
</details>
<details>
  <summary>3 packages failed to build:</summary>
  <ul>
    <li>lua52Packages.sqlite</li>
    <li>lua53Packages.sqlite</li>
    <li>luajitPackages.sqlite</li>
  </ul>
</details>
<details>
  <summary>6 packages built:</summary>
  <ul>
    <li>lua51Packages.sqlite</li>
    <li>lua51Packages.toml</li>
    <li>lua52Packages.toml</li>
    <li>lua53Packages.toml</li>
    <li>lua54Packages.toml</li>
    <li>luajitPackages.toml</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 315732` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>lua54Packages.sqlite</li>
  </ul>
</details>
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>lua51Packages.sqlite</li>
    <li>lua52Packages.sqlite</li>
    <li>lua53Packages.sqlite</li>
    <li>luajitPackages.sqlite</li>
  </ul>
</details>
<details>
  <summary>5 packages built:</summary>
  <ul>
    <li>lua51Packages.toml</li>
    <li>lua52Packages.toml</li>
    <li>lua53Packages.toml</li>
    <li>lua54Packages.toml</li>
    <li>luajitPackages.toml</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 315732` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>lua54Packages.sqlite</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>lua51Packages.sqlite</li>
    <li>lua51Packages.toml</li>
    <li>lua52Packages.sqlite</li>
    <li>lua52Packages.toml</li>
    <li>lua53Packages.sqlite</li>
    <li>lua53Packages.toml</li>
    <li>lua54Packages.toml</li>
    <li>luajitPackages.sqlite</li>
    <li>luajitPackages.toml</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 315732` run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>lua54Packages.sqlite</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>lua51Packages.sqlite</li>
    <li>lua51Packages.toml</li>
    <li>lua52Packages.sqlite</li>
    <li>lua52Packages.toml</li>
    <li>lua53Packages.sqlite</li>
    <li>lua53Packages.toml</li>
    <li>lua54Packages.toml</li>
    <li>luajitPackages.sqlite</li>
    <li>luajitPackages.toml</li>
  </ul>
</details>

**Comment by teto** [comment]
logically the darwin/aarch64-darwin failures shouldn't be new since the code uses the same path, I just changed it to a default. Are you able to confirm if the failures already exist on master please ?

**Review by mrcjkb — APPROVED** [review]

**Comment by GaetanLepage** [comment]
I guess you can go on and merge @teto :)

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 315732` run on aarch64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>lua54Packages.sqlite</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>lua51Packages.sqlite</li>
    <li>lua51Packages.toml</li>
    <li>luaPackages.sqlite (lua52Packages.sqlite)</li>
    <li>luaPackages.toml (lua52Packages.toml)</li>
    <li>lua53Packages.sqlite</li>
    <li>lua53Packages.toml</li>
    <li>lua54Packages.toml</li>
    <li>luajitPackages.sqlite</li>
    <li>luajitPackages.toml</li>
  </ul>
</details>

**Comment by GaetanLepage** [comment]
Result of `nixpkgs-review pr 315732` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package marked as broken and skipped:</summary>
  <ul>
    <li>lua54Packages.sqlite</li>
  </ul>
</details>
<details>
  <summary>9 packages built:</summary>
  <ul>
    <li>lua51Packages.sqlite</li>
    <li>lua51Packages.toml</li>
    <li>luaPackages.sqlite (lua52Packages.sqlite)</li>
    <li>luaPackages.toml (lua52Packages.toml)</li>
    <li>lua53Packages.sqlite</li>
    <li>lua53Packages.toml</li>
    <li>lua54Packages.toml</li>
    <li>luajitPackages.sqlite</li>
    <li>luajitPackages.toml</li>
  </ul>
</details>

File: pkgs/development/lua-modules/overrides.nix [hunk 1]
```diff
@@ -678,9 +678,15 @@ in
     doCheck = true;
     nativeCheckInputs = [ final.plenary-nvim neovim-unwrapped ];
 
+    # the plugin loads the library from either the LIBSQLITE env
+    # or the vim.g.sqlite_clib_path variable.
+    postPatch = ''
+      substituteInPlace lua/sqlite/defs.lua \
+        --replace-fail "path = vim.g.sqlite_clib_path" 'path = vim.g.sqlite_clib_path or  "${sqlite.out}/lib/libsqlite3${stdenv.hostPlatform.extensions.sharedLibrary}"'
```
Comment (SuperSandro2000) [inline pkgs/development/lua-modules/overrides.nix:685]: Shouldn't this rather use lib.getLib?

**Pull request merged**